### ISSUE TITLE
Simplify font setting in usetex mode

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#20289]
### Problem

Currently, setting the font used in usetex mode requires a bit of wrangling with rcParams, e.g.
```python
import matplotlib.pyplot as plt
plt.rcParams.update({
    "text.usetex": True,
    "font.family": "sans-serif",
    "font.sans-serif": ["Helvetica"]})
# for Palatino and other serif fonts use:
plt.rcParams.update({
    "text.usetex": True,
    "font.family": "serif",
    "font.serif": ["Palatino"],
})
```
(per the usetex tutorial).  This is because texmanager checks `font.family`, decides to wrap everything in `\rmfamily` or `\sffamily` based on whether it is "serif", "sans-serif", etc., and then checks `font.serif`/`font.sans-serif`/etc. and calls `\usepackage{helvet}`/`\usepackage{mathpazo}`/etc. which install helvetica as sans-serif font, palatino as serif font, etc. (see https://ctan.org/pkg/psnfss).

### Proposed Solution

A more user-friendly solution would be to directly support setting `font.family = "Helvetica"` for example (consistently with the non-usetex case) and internally translate them as needed to the correct `\{rm,sf}family` call (PSNFSS only supports a small number of fonts anyways so we can hard-code the mapping as needed).

### Additional context and prior art

N/A

-----

Marking as good first issue as this shouldn't need any API discussions and is fairly self-contained, but difficulty: medium as one needs to wrangle a bit with font handling details, especially on tex's side...

**Comment by Carloscerq** [comment]
You mean something like giving the `rc::font.family` the possibility to be one of the fonts of the `TexManager._font_info`?

**Comment by anntzer** [comment]
Yes.

**Comment by dmatos2012** [comment]
Fixed by #20293
